### PR TITLE
Small demon refactor

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

This PR refactors demons a little bit, most notably converts spawn(0) for bloodpool sinking to INVOKE_ASYNC. Also removes implied srcs and cleans up returns a bit.

Tested locally, and all worked without any problems.

Feel free to call this cringe.

## Why It's Good For The Game

Code improvement.

## Changelog
:cl: Arkatos
code: Small demon refactor
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
cringe

File: code/modules/mob/living/bloodcrawl.dm [hunk 1]
```diff
@@ -156,23 +158,23 @@
...
 	if(iscarbon(src))
 		var/mob/living/carbon/C = src
 		for(var/obj/item/bloodcrawl/BC in C)
 			BC.flags_1 = null
 			qdel(BC)
-	qdel(src.holder)
-	src.holder = null
-	return 1
+	qdel(holder)
+	holder = null
```
Comment (nemvar) [inline code/modules/mob/living/bloodcrawl.dm]: Nitpick 1, Swap this out for QDEL_NULL.

File: code/modules/mob/living/bloodcrawl.dm [hunk 2]
```diff
@@ -156,23 +158,23 @@
...
 	B.visible_message("<span class='warning'>[B] starts to bubble...</span>")
 	if(!do_after(src, 20, target = B))
 		return
 	if(!B)
 		return
 	forceMove(B.loc)
-	src.client.eye = src
-	src.visible_message("<span class='warning'><B>[src] rises out of the pool of blood!</B></span>")
+	client.eye = src
+	visible_message("<span class='warning'><B>[src] rises out of the pool of blood!</B></span>")
```
Comment (nemvar) [inline code/modules/mob/living/bloodcrawl.dm]: Nitpick 2, swap the span out for 'boldwarning'

File: code/modules/mob/living/bloodcrawl.dm (hunk 1 above)
Comment (Arkatos1) [reply code/modules/mob/living/bloodcrawl.dm]: Done.

File: code/modules/mob/living/bloodcrawl.dm (hunk 2 above)
Comment (Arkatos1) [reply code/modules/mob/living/bloodcrawl.dm]: Done.

**Review by ghost — APPROVED** [review]
just that 1 thing then this is good

File: code/modules/mob/living/bloodcrawl.dm [hunk 3]
```diff
@@ -124,15 +126,15 @@
 		return TRUE
 
 	to_chat(src, "<span class='danger'>You devour [victim]. Your health is fully restored.</span>")
-	src.revive(full_heal = 1)
+	revive(full_heal = 1)
```
[the remark this answers is not in the document]
Comment (Arkatos1) [reply code/modules/mob/living/bloodcrawl.dm]: Fixed.

File: code/modules/mob/living/bloodcrawl.dm [hunk 4]
```diff
@@ -156,23 +158,22 @@
...
 	B.visible_message("<span class='warning'>[B] starts to bubble...</span>")
 	if(!do_after(src, 20, target = B))
 		return
 	if(!B)
 		return
 	forceMove(B.loc)
-	src.client.eye = src
-	src.visible_message("<span class='warning'><B>[src] rises out of the pool of blood!</B></span>")
+	client.eye = src
+	visible_message("<span class='boldwarning'><B>[src] rises out of the pool of blood!</B></span>")
```
Comment (Couls) [inline code/modules/mob/living/bloodcrawl.dm]: should get rid of the <B> </B> if you're using bold warning

Comment (Arkatos1) [reply code/modules/mob/living/bloodcrawl.dm]: Right, removed.

**Pull request merged**